Execution config factory test

kollerdaniel opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/aliz-ai/jarvis/41)
<!-- Reviewable:end -->

Commit:
create validation test

Commit:
create tests

csabakassai commented on src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java:
```diff
@@ -0,0 +1,77 @@
+
+    @Test
+    public void testValidExecutionActionConfig(){
+        InputStream resourceAsStream = getClass().getResourceAsStream("/sample_tests/test_sql/testSuite.json");
```
use the test files that you have created. 
Also you can name the valid file valid.json

csabakassai commented on src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java:
```diff
@@ -0,0 +1,77 @@
+
+    @Test
+    public void testInvalidParametersExecutionActionConfig(){
+        InputStream resourceAsStream = getClass().getResourceAsStream("/config/invalid.json");
```
the fist 4 line is duplicated in every testcase, exract this common code in a private method

Commit:
Merge branch 'refactor_to_run_from_command_line' into execution_config_factory_test

Commit:
modify the tests

Commit:
Merge branch 'execution_config_factory_test' of https://github.com/aliz-ai/jarvis into execution_config_factory_test

Commit:
Merge remote-tracking branch 'origin/refactor_to_run_from_command_line' into execution_config_factory_test

csabakassai commented on src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java:
```diff
@@ -0,0 +1,78 @@
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    private List<ExecutionActionConfig> Setup(String json){
```
methods  name should start with lowercase letter in Java. 
Also give a more descriptive name to the method, like readExecutionActionConfigs

csabakassai replied:
give the input paramater a more descriptive name
json -> testSuiteJsonPath

csabakassai commented on src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java:
```diff
@@ -0,0 +1,78 @@
+
+    @Test
+    public void testInvalidExecutionActionConfig(){
+        InputStream resourceAsStream = getClass().getResourceAsStream("/sample_tests/test_sql/invalid.json");
```
here you do not test any code from the application so this test does not needed

csabakassai commented on src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java:
```diff
@@ -0,0 +1,78 @@
+
+    @Test
+    public void testInvalidParametersExecutionActionConfig(){
+        List<ExecutionActionConfig> executionActionConfigs = this.Setup("/config/invalid.json");
```
How the code should work:
1. Parse the ExecutionActionConfig
2. Determine the executionType 
3. Parse and validate the ExecutionActionConfig based on the executionType, the exeuctionType is required
a, If BqQuery
- required params: queryPath, executionContext
- validate if there is a local context
- validate  if the queryPath is an existing file
- validate if the exeuctionContext exists
b. any other executionType, throw an unsupportedoperation Execption

Write a unit test for all of the above here, before you do the implementation

nbali committed:
Merge branch 'master' into execution_config_factory_test

Commit:
Merge branch 'master' of https://github.com/aliz-ai/jarvis into execution_config_factory_test

Commit:
Merge branch 'execution_config_factory_test' of https://github.com/aliz-ai/jarvis into execution_config_factory_test

Commit:
edit execution action tests

Commit:
edit execution action tests

nbali reviewed (changes requested):
> Reviewed 1 of 3 files at r2, 1 of 2 files at r3, 2 of 3 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 17 unresolved discussions (waiting on @kollerdaniel)

___
*[examples/sample_01/contexts.json, line 14 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4ITb7Cerp5vM4ArzF:-Mx4ITb7Cerp5vM4ArzG:b-2sykqr) ([raw file](https://github.com/aliz-ai/jarvis/blob/991e0e3f6974ba354d00762f4e2ac75b5247550a/examples/sample_01/contexts.json#L14)):*
> ```JSON
>     "contextType": "LocalContext",
>     "parameters": {
>       "repositoryRoot": "/Users/csabakassai/aliz/aliz-test-runner/repository/jarvis"
> ```

this file shouldn't be modified at all (most likely)
___
*[src/main/java/ai/aliz/jarvis/util/JarvisUtil.java, line 3 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4IXkf3QJntsnRRGxc:-Mx4IXkf3QJntsnRRGxd:b-m27nif) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/main/java/ai/aliz/jarvis/util/JarvisUtil.java#L3)):*
> ```Java
> package ai.aliz.jarvis.util;
> 
> import ai.aliz.jarvis.context.TestContext;
> ```

is this needed?
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 13 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4JTtREtjE_vUtXfkt:-Mx4JTtREtjE_vUtXfku:b-sdikar) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L13)):*
> ```Java
> import ai.aliz.jarvis.context.TestContextType;
> import ai.aliz.jarvis.testconfig.ExecutionActionConfig;
> import ai.aliz.jarvis.testconfig.ExecutionActionConfigFactory;
> ```

both this and the InitActionConfigFactory tests should be in the same package as the tested classes
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 39 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4IaCQ5M1rMzJrdwSi:-Mx4IaCQ5M1rMzJrdwSj:b-hf3ojs) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L39)):*
> ```Java
> @RunWith(SpringJUnit4ClassRunner.class)
> @SpringBootTest
> //@TestPropertySource(properties = "context=src/test/resources/test_context.json")
> ```

most likely not needed if `TestContextLoader ` is a mock
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 64 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4IjpK9ZRLNaEkBfjY:-Mx4IjpK9ZRLNaEkBfjZ:bz76aca) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L64)):*
> ```Java
>         return executionActionConfigs;
>     }
> ```

create method to check if it loaded 1, and only return that 1 execution action config
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 67 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4JQQC4NvEUCfm_XN8:-Mx4JQQC4NvEUCfm_XN9:b-ujn01z) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L67)):*
> ```Java
>     @Test
>     public void testValidExecutionActionConfig(){
>         List<ExecutionActionConfig> executionActionConfigs = this.readExecutionActionConfigs("/config/valid.json");
> ```

/config/ should be /execution/
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 73 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4IrVtCZH7GI4GNtbS:-Mx4IrVtCZH7GI4GNtbT:b-yz10gr) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L73)):*
> ```Java
>         assertThat(executionActionConfig.getType(), is(ExecutionType.BqQuery));
>         assertNull(executionActionConfig.getDescriptorFolder());
>         assertThat(executionActionConfig.getExecutionContext(), is("test_bq"));
> ```

assert any remaining property from the valid/happy path
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 80 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4J-BU9KgPWnDz6Zjq:-Mx4J-BU9KgPWnDz6Zjr:bmc59c7) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L80)):*
> ```Java
>         List<ExecutionActionConfig> executionActionConfigs = this.readExecutionActionConfigs("/config/valid.json");
>         ExecutionActionConfig executionActionConfig = executionActionConfigs.get(0);
>         executionActionConfig.setType(ExecutionType.valueOf("BqQuery"));
> ```

setType should never be called in these tests
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 83 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4J2wi1tlHgqw1fx99:-Mx4J2wi1tlHgqw1fx9A:b-mvarh5) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L83)):*
> ```Java
>         assertThat(executionActionConfig.getProperties().get("sourcePath"), is("test/project/src/test/resources/sample_tests/test_sql/test.sql"));
>         assertThat(executionActionConfig.getExecutionContext(), is("test_bq"));
>     }
> ```

test invalid* jsons
___
*[src/test/resources/config/invalid.json, line 1 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4JFwn7kYmk54-Hn6D:-Mx4JFwo5vqqDHLRUc8d:b-699x31) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/resources/config/invalid.json#L1)):*
> ```JSON
> {
> ```

should be inside src/test/resources/execution folder
___
*[src/test/resources/config/invalid.json, line 1 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4JGcZABzBni3hTSwb:-Mx4JGcZABzBni3hTSwc:b4e28vf) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/resources/config/invalid.json#L1)):*
> ```JSON
> {
> ```

describe why it's invalid in the name
___
*[src/test/resources/config/valid.json, line 1 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4JCKi2j1rWc5_RP5O:-Mx4JCKi2j1rWc5_RP5P:b-699x31) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/resources/config/valid.json#L1)):*
> ```JSON
> {
> ```

should be inside src/test/resources/execution folder


<!-- Sent from Reviewable.io -->

nbali reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 17 unresolved discussions (waiting on @csabakassai and @kollerdaniel)

___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 65 at r3](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MqyBWBHTl6m7IPswnxo-r3-65:-Mx4K5B5CBqK8ETHFiQW:b3kqx) ([raw file](https://github.com/aliz-ai/jarvis/blob/8a21a92979ab3f55a2722ad2d71716313a64aed3/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L65)):*
<details><summary><i>Previously, csabakassai (Kassai Csaba) wrote…</i></summary><blockquote>

here you do not test any code from the application so this test does not needed
</blockquote></details>

.
___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 73 at r3](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MqyBWBHTl6m7IPswnxo-r3-73:-Mx4JszO69vFVwy5akec:brjgacl) ([raw file](https://github.com/aliz-ai/jarvis/blob/8a21a92979ab3f55a2722ad2d71716313a64aed3/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L73)):*
<details><summary><i>Previously, csabakassai (Kassai Csaba) wrote…</i></summary><blockquote>

How the code should work:
1. Parse the ExecutionActionConfig
2. Determine the executionType 
3. Parse and validate the ExecutionActionConfig based on the executionType, the exeuctionType is required
a, If BqQuery
- required params: queryPath, executionContext
- validate if there is a local context
- validate  if the queryPath is an existing file
- validate if the exeuctionContext exists
b. if NoOps no validation needed
c. any other executionType, throw an unsupportedoperation Execption

Write a unit test for all of the above here, before you do the implementation

</blockquote></details>

just for clarification  - all of these should be inside the business code itself, and the junit test should only validate if there are exceptions being thrown if the provided json file has something wrong/missing


<!-- Sent from Reviewable.io -->

Commit:
edit execution action tests

Commit:
edit execution action tests

nbali reviewed (changes requested):
> Reviewed 10 of 11 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: 10 of 12 files reviewed, 28 unresolved discussions (waiting on @csabakassai, @kollerdaniel, and @nbali)

___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 63 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFKngP98qH8068xkbH:-MxFKngP98qH8068xkbI:b96ujmi) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L63)):*
> ```Java
>         List<ExecutionActionConfig> executionActionConfig = executionActionConfigFactory.getExecutionActionConfig(map);
>         return executionActionConfig;
>     }
> ```

don't duplicate code, extract this into a method with an additional parameter, and just call it differently

so 3 method instead of 2
- extracted method
- read calling 1st
- read with bad context calling 1st
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 65 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFLE3K1vD8fbp8B6Xc:-MxFLE3K1vD8fbp8B6Xd:b-o2olym) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L65)):*
> ```Java
>     }
> 
>     private ExecutionActionConfig getOneExecutionActionConfig(String testSuiteJsonPath){
> ```

single, singular, only is better fitting the "get one" as that doesn't explain that we need the _only one_
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 67 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFL7YnCnpid1_SnT85:-MxFL7YnCnpid1_SnT86:b-bwp0gy) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L67)):*
> ```Java
>     private ExecutionActionConfig getOneExecutionActionConfig(String testSuiteJsonPath){
>         List<ExecutionActionConfig> executionActionConfigs = this.readExecutionActionConfigs(testSuiteJsonPath);
>         ExecutionActionConfig executionActionConfig = executionActionConfigs.get(0);
> ```

if you have read 0 this will fail with an exception with crappy description
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 71 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFLPUMDbahsEZ_vOAC:-MxFLPUN4dAFr3Ul6KaH:b-eng735) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L71)):*
> ```Java
>             return executionActionConfig;
>         }else{
>             return null;
> ```

I'm pretty sure you don't want to return with zero,  and the tests should fail if that happens with a proper error msg

you can use a stream with MoreCollectors.onlyElement() collector, or Iterables.getOnlyElement on the list for example
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 82 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFLwMD3k0FJrVq_WZm:-MxFLwMD3k0FJrVq_WZn:bnj7hl0) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L82)):*
> ```Java
>         }else{
>             return null;
>         }
> ```

same as above
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 100 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFM4kfAhecdOsfNc5t:-MxFM4kfAhecdOsfNc5u:b6odcy9) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L100)):*
> ```Java
>         assertThat(executionActionConfig.getProperties().get("sourcePath"), is("test/project/src/test/resources/sample_tests/test_sql/test.sql"));
>         assertThat(executionActionConfig.getExecutionContext(), is("test_bq"));
>     }
> ```

what is the purpose of this test? the previous covers everything already
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 103 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFMRqn-NgO1nRDne6j:-MxFMRqn-NgO1nRDne6k:b-xr8qa2) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L103)):*
> ```Java
> 
>     @Test
>     public void testOtherExecutionType(){
> ```

if you want to test other but valid execution types (like noop, or airflow) you need new jsons with that being used
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 105 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFMCsKDVbEKR_g82Eg:-MxFMCsKDVbEKR_g82Eh:bxh6mp4) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L105)):*
> ```Java
>     public void testOtherExecutionType(){
>         ExecutionActionConfig executionActionConfig = getOneExecutionActionConfig("/execution/valid.json");
>         exceptionRule.expect(AssertionError.class);
> ```

you never expect an assertion error, the expected exception is to except something thrown from the business code and not by your own check
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 112 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFMekk7n3E0IGW--Zj:-MxFMekk7n3E0IGW--Zk:b-nlq9rn) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L112)):*
> ```Java
>     public void testWithoutQueryPath(){
>         ExecutionActionConfig executionActionConfig = getOneExecutionActionConfig("/execution/invalidWithoutQueryPath.json");
>         exceptionRule.expect(AssertionError.class);
> ```

see above

this is required before the actual parsin so before the getOne...
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 124 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFMoGuBbD4iVdczrlQ:-MxFMoGuBbD4iVdczrlR:b1r1jq1) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L124)):*
> ```Java
>     @Test
>     public void testWithoutExecutionContext(){
>         exceptionRule.expect(NullPointerException.class);
> ```

I wouldn't specify it, a runtimeexception will do just fine
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 131 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFMzLS7h7XwH7sq5MU:-MxFMzLS7h7XwH7sq5MV:byr4ffv) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L131)):*
> ```Java
>     public void testBadQueryPath(){
>         ExecutionActionConfig executionActionConfig = getOneExecutionActionConfig("/execution/invalidBadQueryPath.json");
>         exceptionRule.expect(AssertionError.class);
> ```

see above
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 144 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFN72S1JjsRs90GCgo:-MxFN72S1JjsRs90GCgp:byr4ffv) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L144)):*
> ```Java
>     public void testBadExecutionContext(){
>         ExecutionActionConfig executionActionConfig = getOneExecutionActionConfig("/execution/invalidBadExecutionContext.json");
>         exceptionRule.expect(AssertionError.class);
> ```

see above
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 150 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFN9Ob7-lYOGj8mw2F:-MxFN9Ob7-lYOGj8mw2G:be36gma) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L150)):*
> ```Java
>     @Test
>     public void testReadExecutionActionConfigsWithoutLocalContext(){
>         exceptionRule.expect(NullPointerException.class);
> ```

Runtimeexception is enough
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 157 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFNJgo5Kqhc_DgoSk6:-MxFNJgo5Kqhc_DgoSk7:bbw4h1b) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L157)):*
> ```Java
>     public void testNoOpsExecutionType(){
>         ExecutionActionConfig executionActionConfig = getOneExecutionActionConfig("/execution/valid.json");
>         exceptionRule.expect(AssertionError.class);
> ```

see above
___
*[src/test/resources/execution/invalidBadExecutionType.json, line 7 at r7](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFJVfPBY92Ahk1zvf2:-MxFJVfPBY92Ahk1zvf3:b-dfq50p) ([raw file](https://github.com/aliz-ai/jarvis/blob/ca49b8e78c50952fb3d95e7fe78db7e51b330a3d/src/test/resources/execution/invalidBadExecutionType.json#L7)):*
> ```JSON
>       "executionType": "SQL",
>       "queryPath": "/src/test/resources/sample_tests/test_sql/test.sql",
>       "executionContext": "valid_bq"
> ```

it's invalid_bq everywhere, why the distinction? or maybe every - except one - invalid_bq is wrong and they all should be valid_bq?


<!-- Sent from Reviewable.io -->

nbali reviewed (commented):
> Reviewed 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: 11 of 12 files reviewed, 28 unresolved discussions (waiting on @csabakassai, @kollerdaniel, and @nbali)



<!-- Sent from Reviewable.io -->

Commit:
edit execution action config factory tests

Commit:
edit execution action config factory tests

nbali reviewed (changes requested):
> Reviewed 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: 12 of 13 files reviewed, 24 unresolved discussions (waiting on @csabakassai, @kollerdaniel, and @nbali)

___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 65 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFLE3K1vD8fbp8B6Xc:-MyNf43jBkErOQzx4R6Z:bxekp6k) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L65)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

single, singular, only is better fitting the "get one" as that doesn't explain that we need the _only one_
</blockquote></details>

I meant getSingle, getSingular, getOnly, etc method names usually makes more sense with a verb (I mean @see getOnlyElement :P )
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 67 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFL7YnCnpid1_SnT85:-MyNfLnf1n1LLS87yvfb:brwy0cs) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L67)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

if you have read 0 this will fail with an exception with crappy description
</blockquote></details>

still, you kept this line
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 82 at r6](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFLwMD3k0FJrVq_WZm:-MyNfgdT2pmGPMBNQon0:bnj7hl0) ([raw file](https://github.com/aliz-ai/jarvis/blob/26bf582d25ad069fd75c3db23aec248f4b4a82ea/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L82)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

same as above
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 57 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNdwpI1BJDKwdhK9Mo:-MyNdwpI1BJDKwdhK9Mp:b-pl4rew) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L57)):*
> ```Java
>     }
> 
>     private List<ExecutionActionConfig> readExecutionActionConfigs(String testSuiteJsonPath){
> ```

when you name a method, try to think with the mindset to name it that it's obvious what it does without looking up the implementation

right now you have two method that is essentially same, but one uses a local context, meanwhile at the other it's configurable

I would be fine having the same name, just different signature (that would indicate the difference), I would be also fine with having an even more descriptive name, but right now... what is the difference between `readExecutionActionConfigs` and `˙readExecutionAction` that justifies the different name?

I mean I get that your aim was that that method should be called only from the next two methods, but if you are writing a test, there is no indication which one to choose
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 71 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNff0x3NAESUz_xO76:-MyNff0x3NAESUz_xO77:b-6n0vj3) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L71)):*
> ```Java
>     }
> 
>     private ExecutionActionConfig singleExecutionActionConfigWithBadContext(String testSuiteJsonPath){
> ```

get prefix
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 90 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNfpzsDUu8ot1dPkBW:-MyNfpzsDUu8ot1dPkBX:b-lmsvrj) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L90)):*
> ```Java
>         exceptionRule.expect(RuntimeException.class);
>         ExecutionActionConfig executionActionConfig = singleExecutionActionConfig("/execution/airflow.json");
>         assertThat(executionActionConfig.getType(), is("Airflow"));
> ```

the code execution won't reach this statement, but don't you ahve an `ExecutionType` anum for this value? and enum will never be equal with a string anyway
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 95 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNgM3i8P7MnZp0Wa0P:-MyNgM3i8P7MnZp0Wa0Q:bemsss) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L95)):*
> ```Java
>     @Test
>     public void testWithoutQueryPath(){
>         exceptionRule.expect(RuntimeException.class);
> ```

fyi if you want to be sure that issues aren't triggering each other you can also test for exception message
___
*[src/test/resources/execution/invalidBadExecutionType.json, line 7 at r7](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MxFJVfPBY92Ahk1zvf2:-MyNh9e-5veBQF1Z1w5r:b-layz4t) ([raw file](https://github.com/aliz-ai/jarvis/blob/ca49b8e78c50952fb3d95e7fe78db7e51b330a3d/src/test/resources/execution/invalidBadExecutionType.json#L7)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

it's invalid_bq everywhere, why the distinction? or maybe every - except one - invalid_bq is wrong and they all should be valid_bq?
</blockquote></details>

--> "or maybe every - except one - invalid_bq is wrong and they all should be test_bq ?"
___
*[src/test/resources/execution/invalidBadQueryPath.json, line 7 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNhG86AyFx1bTjKMpD:-MyNhG86AyFx1bTjKMpE:bh8dl6x) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/resources/execution/invalidBadQueryPath.json#L7)):*
> ```JSON
>       "executionType": "BqQuery",
>       "queryPath": "/src/test/resources/sample_tests/test_sql/invalid.sql",
>       "executionContext": "invalid_bq"
> ```

test_bq 
___
*[src/test/resources/execution/invalidWithoutExecutionType.json, line 6 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNhIqu5NLJxUyejZn_:-MyNhIqu5NLJxUyejZna:bh8dl6x) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/resources/execution/invalidWithoutExecutionType.json#L6)):*
> ```JSON
>     {
>       "queryPath": "/src/test/resources/sample_tests/test_sql/test.sql",
>       "executionContext": "invalid_bq"
> ```

test_bq 
___
*[src/test/resources/execution/invalidWithoutQueryPath.json, line 6 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNhKNKB1AnB61fduBY:-MyNhKNKB1AnB61fduBZ:bh8dl6x) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/resources/execution/invalidWithoutQueryPath.json#L6)):*
> ```JSON
>     {
>       "executionType": "BqQuery",
>       "executionContext": "invalid_bq"
> ```

test_bq 
___
*[src/test/resources/execution/invalidBadExecutionContext.json, line 7 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNgzsA0n7OWjRo9YMz:-MyNgzsA0n7OWjRo9YN-:b8cgzt) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/resources/execution/invalidBadExecutionContext.json#L7)):*
> ```JSON
>       "executionType": "BqQuery",
>       "queryPath": "/src/test/resources/sample_tests/test_sql/test.sql",
>       "executionContext": "invalid_bq"
> ```

this is the ONLY test that should use this executionContext, others should use a valid one ( test_bq )


<!-- Sent from Reviewable.io -->

nbali reviewed (changes requested):
> Reviewed 1 of 11 files at r6.
> Dismissed @csabakassai from 4 discussions.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 15 unresolved discussions (waiting on @csabakassai and @kollerdaniel)

___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 138 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNk8Su5EUvMJTwO5mK:-MyNk8Su5EUvMJTwO5mL:b-j3weyh) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L138)):*
> ```Java
>     public void testAirFlowExecutionType(){
>         exceptionRule.expect(RuntimeException.class);
>         ExecutionActionConfig executionActionConfig = singleExecutionActionConfig("/execution/valid.json");
> ```

I'm pretty sure that is not what you intended


<!-- Sent from Reviewable.io -->

Commit:
edit execution action config factory tests

nbali reviewed (changes requested):
> Reviewed 1 of 1 files at r9, 5 of 5 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 6 unresolved discussions (waiting on @csabakassai and @kollerdaniel)

___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 57 at r8](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MyNdwpI1BJDKwdhK9Mo:-Mynk7Rg0pKqNC515sg8:b-pywf85) ([raw file](https://github.com/aliz-ai/jarvis/blob/a1c8743b6080be1214abbd43c6e746edf043f103/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L57)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

when you name a method, try to think with the mindset to name it that it's obvious what it does without looking up the implementation

right now you have two method that is essentially same, but one uses a local context, meanwhile at the other it's configurable

I would be fine having the same name, just different signature (that would indicate the difference), I would be also fine with having an even more descriptive name, but right now... what is the difference between `readExecutionActionConfigs` and `˙readExecutionAction` that justifies the different name?

I mean I get that your aim was that that method should be called only from the next two methods, but if you are writing a test, there is no indication which one to choose
</blockquote></details>

okay, I will let this slide, but now the method name is like readWithFoo, read "what"? :)
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 65 at r10](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MynkbnR5QXRQ0lqa40G:-MynkbnR5QXRQ0lqa40H:btirnfj) ([raw file](https://github.com/aliz-ai/jarvis/blob/b7f0916076b77026152b0c30d33c178a66944310/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L65)):*
> ```Java
>     }
> 
>     private ExecutionActionConfig getSingleExecutionActionConfig(String testSuiteJsonPath){
> ```

fyi the called method has context info, this doesnt, the next getSingle does, so the does the called method  try to be consistent next time
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 70 at r10](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MynkSpzAhO8pHli1oXV:-MynkSq-0SlVAKvR8In-:b-ctctnu) ([raw file](https://github.com/aliz-ai/jarvis/blob/b7f0916076b77026152b0c30d33c178a66944310/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L70)):*
> ```Java
>     }
> 
>     private ExecutionActionConfig getSingleExecutionActionConfigWithBadContext(String testSuiteJsonPath){
> ```

fyi bad context, invalid context, etc try to be consistent next time
___
*[src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java, line 88 at r10](https://reviewable.io/reviews/aliz-ai/jarvis/41#-MynkrN87lrK6S4fXMKC:-MynkrN87lrK6S4fXMKD:b-krqjc2) ([raw file](https://github.com/aliz-ai/jarvis/blob/b7f0916076b77026152b0c30d33c178a66944310/src/test/java/ai/aliz/jarvis/testconfig/TestExecutionActionConfigFactory.java#L88)):*
> ```Java
>         exceptionRule.expect(RuntimeException.class);
>         ExecutionActionConfig executionActionConfig = getSingleExecutionActionConfig("/execution/airflow.json");
>         assertThat(executionActionConfig.getType(), is(ExecutionType.Airflow));
> ```

there is no way this line will be reached


<!-- Sent from Reviewable.io -->

nbali committed:
PR CR

nbali committed:
PR CR

nbali reviewed (commented):
> Reviewed 1 of 1 files at r11, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 3 unresolved discussions (waiting on @csabakassai and @kollerdaniel)



<!-- Sent from Reviewable.io -->

nbali committed:
Pinning proper GH action version

Error: On 2022-04-05, the default branch will be renamed from "master" to "main". Your action is currently pinned to "@master". Even though GitHub creates redirects for renamed branches, testing found that this rename breaks existing GitHub Actions workflows that are pinned to the old branch name.

We strongly advise updating your GitHub Action YAML from:
    uses: 'google-github-actions/setup-gcloud@master'
to:
    uses: 'google-github-actions/setup-gcloud@v0'

nbali reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: 13 of 14 files reviewed, 3 unresolved discussions (waiting on @csabakassai and @kollerdaniel)

*[a discussion](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mynp8g883vENJn5E2g5:-Mynp8g94hs2-n6L8KuO:bs9desr) (no related file):*
@kollerdaniel I just realized test_bq is nowhere to be found in the loading itself, so that should be mocked as well, not only the local/invalid context

---


<!-- Sent from Reviewable.io -->

nbali reviewed (commented):
> Reviewed 1 of 1 files at r13.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed (commit messages unreviewed), 3 unresolved discussions (waiting on @csabakassai and @kollerdaniel)



<!-- Sent from Reviewable.io -->

nbali reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 3 unresolved discussions (waiting on @csabakassai and @kollerdaniel)



<!-- Sent from Reviewable.io -->

Commit:
edit execution action config factory tests

Commit:
Merge branch 'execution_config_factory_test' of https://github.com/aliz-ai/jarvis into execution_config_factory_test

nbali reviewed (commented):
> Reviewed 1 of 3 files at r14, 3 of 3 files at r15, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 3 unresolved discussions (waiting on @csabakassai and @kollerdaniel)



<!-- Sent from Reviewable.io -->

nbali reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 1 unresolved discussion (waiting on @csabakassai)

___
*[src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java, line 73 at r5](https://reviewable.io/reviews/aliz-ai/jarvis/41#-Mx4IrVtCZH7GI4GNtbS:-N-TK5ziB_WAuWUi6BQ0:b-xwf1e9) ([raw file](https://github.com/aliz-ai/jarvis/blob/247e24396aee37caf6f47c23f98aa95937c150eb/src/test/java/ai/aliz/jarvis/service/executionaction/TestExecutionActionConfigFactory.java#L73)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

assert any remaining property from the valid/happy path
</blockquote></details>

moved to another comment


<!-- Sent from Reviewable.io -->

Commit:
refactor execution action config factory

nbali reviewed (changes requested):
> Reviewed 1 of 1 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 9 unresolved discussions (waiting on @csabakassai and @kollerdaniel)

___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 40 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-Xnb3s0P_UItUeX7sy:-N-Xnb3s0P_UItUeX7sz:bxd7usl) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L40)):*
> ```Java
>     
>     private List<ExecutionActionConfig> getExecutionActionConfigs(List<Map<String, String>> executions) {
>         List<ExecutionActionConfig> executionActionConfigs = Lists.newArrayList();
> ```

this should be inside the flatMap as it's being reused during the multiple flatMap calls
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 44 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-XnlcN9TA0eOr8RCyW:-N-XnlcN9TA0eOr8RCyX:b-svlype) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L44)):*
> ```Java
>         String repositoryRoot = moderateFilePathSlashes(contextLoader.getContext("local").getParameter("repositoryRoot"));
>         return executions.stream()
>                          .flatMap(e -> {
> ```

this should be a `.map(` actually no need for flatmap, just complicates stuff
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 46 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-Xm-ahDDJ_xp8X_eE2:-N-Xm-aiDRYrHuPEYUWA:b-uwbtdr) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L46)):*
> ```Java
>                          .flatMap(e -> {
>                              ExecutionActionConfig executionActionConfig = new ExecutionActionConfig();
>                              switch (ExecutionType.valueOf(checkExecutionType(e.get("executionType")))){
> ```

don't calculate values multiple times, extract it to a variable

method calls could be:
- costly computation wise
- flaky with different result
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 48 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-XmFA30vxL2Yv3niZ7:-N-XmFA4B04TK_e8THLf:b-s0swmb) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L48)):*
> ```Java
>                              switch (ExecutionType.valueOf(checkExecutionType(e.get("executionType")))){
>                                  case BqQuery:
>                                      if (e.get("queryPath") == null || !e.get("queryPath").substring(e.get("queryPath").length() - 8).equals("test.sql") || !e.get("executionContext").equals("test_bq")){
> ```

where does this come from? comparison to test.sql? is there any requirement anywhere about this?
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 48 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-XmLWh1mhyZysXY5Mc:-N-XmLWh1mhyZysXY5Md:b-fih7wf) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L48)):*
> ```Java
>                              switch (ExecutionType.valueOf(checkExecutionType(e.get("executionType")))){
>                                  case BqQuery:
>                                      if (e.get("queryPath") == null || !e.get("queryPath").substring(e.get("queryPath").length() - 8).equals("test.sql") || !e.get("executionContext").equals("test_bq")){
> ```

test_bq is a test related data, shouldn't appear in this class at all
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 49 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-XmQDCEfGns4z40vqn:-N-XmQDCEfGns4z40vqo:b-k0owtk) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L49)):*
> ```Java
>                                  case BqQuery:
>                                      if (e.get("queryPath") == null || !e.get("queryPath").substring(e.get("queryPath").length() - 8).equals("test.sql") || !e.get("executionContext").equals("test_bq")){
>                                          throw new RuntimeException();
> ```

never throw any exception without proper message/context what it's being thrown for
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 59 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-Xn5G69F9lClNpDwS5:-N-Xn5G73TfPodIjNQcF:b-i5ft8c) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L59)):*
> ```Java
>                                  case NoOps:
>                                  case Talend:
>                                  case Airflow:
> ```

`default:` is missing
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 60 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-XmYim8GIghM84y3DT:-N-XmYim8GIghM84y3DU:b-yx0u5w) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L60)):*
> ```Java
>                                  case Talend:
>                                  case Airflow:
>                                     throw new RuntimeException();
> ```

UnsupportedOperationException or IllegalArgument/IllegalState exception with proper message


<!-- Sent from Reviewable.io -->

Commit:
refactor execution action config factory

nbali reviewed (changes requested):
> Reviewed 1 of 1 files at r17, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 18 unresolved discussions (waiting on @csabakassai and @kollerdaniel)

___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 46 at r16](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N-Xm-ahDDJ_xp8X_eE2:-N025cdaDMgGsggU6MLe:b-p41bti) ([raw file](https://github.com/aliz-ai/jarvis/blob/a6e885c438add293d61b4d957266e26c1f3773da/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L46)):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

don't calculate values multiple times, extract it to a variable

method calls could be:
- costly computation wise
- flaky with different result
</blockquote></details>

you still do it for this variable later on
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 11 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N0269Do8t5xrQmj4epQ:-N0269Do8t5xrQmj4epR:bugsbtb) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L11)):*
> ```Java
> import java.util.stream.Collectors;
> 
> import com.google.api.gax.rpc.InvalidArgumentException;
> ```

this is a very specific exception of the gax-rpc library, use the generic 'IllegalArgumentException'
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 44 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N025WZcECC5Ef-kEJXa:-N025WZcECC5Ef-kEJXb:bck8da3) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L44)):*
> ```Java
>         String repositoryRoot = moderateFilePathSlashes(contextLoader.getContext("local").getParameter("repositoryRoot"));
>         return executions.stream().map(e -> {
>                     List<ExecutionActionConfig> executionActionConfigs;
> ```

don't keep this list, it's useless now
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 49 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N026YTW3lCkl5ZH4Nxm:-N026YTW3lCkl5ZH4Nxn:bpp6stj) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L49)):*
> ```Java
>                     switch (executionType){
>                         case BqQuery:
>                             executionActionConfigs =  Lists.newArrayList();
> ```

not needed
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 50 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N026hFP-DlBrl3bQMCV:-N026hFP-DlBrl3bQMCW:b-aanj20) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L50)):*
> ```Java
>                         case BqQuery:
>                             executionActionConfigs =  Lists.newArrayList();
>                             String newQueryPathName = "." + e.get("queryPath");
> ```

if they use `foo/bar` you just made it into an invalid format `.foo/bar`, or if they use `./foo/bar` you just moved it up by a folder

anyway the reason this isn't okay is that if you check how the code actually works a few lines later, the queryPath is being used together with the 'repositoryRoot' without that it's a relative path without an actual root

you should calculate the "sourcePath"/"SOURCE_PATH", check if it exists and if yes set the property, if not fail the load process
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 52 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N02885n8lqnlS481y_e:-N02885n8lqnlS481y_f:b-cgtc57) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L52)):*
> ```Java
>                             String newQueryPathName = "." + e.get("queryPath");
>                             File queryPath = new File(newQueryPathName);
>                             String executionContext = e.get("executionContext");
> ```

you already have a requireNonNull check surrounding this, you should move that check here, as you are validating now afterall
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 53 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N028Hn_AU4QdREV-_rS:-N028Hn_AU4QdREV-_rT:b-wiz51j) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L53)):*
> ```Java
>                             File queryPath = new File(newQueryPathName);
>                             String executionContext = e.get("executionContext");
>                             if (contextLoader == null){
> ```

this can't be null, you just used 'context' a few lines above, the intellij should warn you about this..
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 54 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N028RFtEt7jQA6lg4wn:-N028RFtEt7jQA6lg4wo:bcvvv4g) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L54)):*
> ```Java
>                             String executionContext = e.get("executionContext");
>                             if (contextLoader == null){
>                                 Objects.requireNonNull(e, "Context must not be null.");
> ```

this will never fail, 'e' will never be null
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 57 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N027yP53Dbhmheh7FrU:-N027yP53Dbhmheh7FrV:b-4y7rcz) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L57)):*
> ```Java
>                             }
>                             else if (!queryPath.isFile()){
>                                 throw new NullPointerException("This query path doesn't exist.");
> ```

what query path, help the used to figure it out, include it into the exception msg
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 59 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N0298Ef22bZlCaC2ayI:-N0298Ef22bZlCaC2ayJ:b-ycj6hj) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L59)):*
> ```Java
>                                 throw new NullPointerException("This query path doesn't exist.");
>                             }
>                             else if (executionContext.isEmpty()){
> ```

this is not null, it's an empty string, if it's null you fail here with a null pointer exception without any proper msg

also the point for executionContext isn't only to check if it exists, and it's not an empty string, but that the context actually exists, so 'contextLoader.getContext(' can return an existing context
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 60 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N0298o1E4lX2aziDyOx:-N0298o1E4lX2aziDyOy:b-man968) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L60)):*
> ```Java
>                             }
>                             else if (executionContext.isEmpty()){
>                                 Objects.requireNonNull(e, "Execution context must not be null.");
> ```

e will never be null, so this will never happen
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 63 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N025qekC1KI14b245ll:-N025qekC1KI14b245lm:b9i7zt) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L63)):*
> ```Java
>                             }
>                             else if (!existsAValidExecutionContext(queryPath, executionContext)){
>                                 throw new IllegalArgumentException("Invalid execution type or illegal folder syntax.");
> ```

which one, what is invalid, try to think with the mindset of the user
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 65 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N029b8G3CYEKE_9KZj2:-N029b8G3CYEKE_9KZj3:b-ifg18t) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L65)):*
> ```Java
>                                 throw new IllegalArgumentException("Invalid execution type or illegal folder syntax.");
>                             }else{
>                                 executionActionConfig.setType(ExecutionType.valueOf(checkExecutionType(e.get("executionType"))));
> ```

you already calculated before
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 66 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N029cWF1CSj-cPKuyyc:-N029cWF1CSj-cPKuyyd:buafeuh) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L66)):*
> ```Java
>                             }else{
>                                 executionActionConfig.setType(ExecutionType.valueOf(checkExecutionType(e.get("executionType"))));
>                                 executionActionConfig.getProperties().put(SOURCE_PATH, repositoryRoot + e.get("queryPath"));
> ```

you should ahve already calculated before for the validation
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 67 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N029fuX2ZerlzjOeQfZ:-N029fuX2ZerlzjOeQf_:buafeuh) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L67)):*
> ```Java
>                                 executionActionConfig.setType(ExecutionType.valueOf(checkExecutionType(e.get("executionType"))));
>                                 executionActionConfig.getProperties().put(SOURCE_PATH, repositoryRoot + e.get("queryPath"));
>                                 executionActionConfig.setExecutionContext(Objects.requireNonNull(e.get("executionContext"), "executionContext property must be specified on BqQuery executions"));
> ```

you should ahve already calculated before for the validation
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 81 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N029lT24t6C7S4gfX-z:-N029lT24t6C7S4gfX0-:b31z2im) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L81)):*
> ```Java
>     }
> 
>     private boolean existsAValidExecutionContext(File queryPath, String executionContext){
> ```

this is not the way, you have a `contextLoader.getContext()` use that
___
*[src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java, line 110 at r17](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N026PG16ZJnWnAOWVIk:-N026PG16ZJnWnAOWVIl:b-stlius) ([raw file](https://github.com/aliz-ai/jarvis/blob/2702bcffd3c563b894bb63217472b50b5900e319/src/main/java/ai/aliz/jarvis/testconfig/ExecutionActionConfigFactory.java#L110)):*
> ```Java
>     }
>     
>     private String checkExecutionType(String executionType) {
> ```

fyi this could return the found `ExecutionType` no need to look it up externally again


<!-- Sent from Reviewable.io -->

nbali committed:
Enriched unit test contexts with contextId also replaced path comparison with canonical path comparison

nbali committed:
Added required checks to ExecutionActionConfigFactory

nbali committed:
Merge branch 'master' into execution_config_factory_test

Conflicts:
	src/main/java/ai/aliz/jarvis/context/JarvisContextLoader.java
	src/test/java/ai/aliz/jarvis/testconfig/TestInitActionConfigFactory.java

nbali reviewed (dismissed):
> Reviewed 1 of 1 files at r18, 4 of 4 files at r19, 9 of 9 files at r20, all commit messages.
> Dismissed @csabakassai from a discussion.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @csabakassai)



<!-- Sent from Reviewable.io -->

nbali committed:
Fixed packages

nbali reviewed (approved):
> Reviewed 2 of 2 files at r21, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @csabakassai)



<!-- Sent from Reviewable.io -->

nbali reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 1 unresolved discussion (waiting on @csabakassai, @gaborvolczer, and @kollerdaniel)

*[a discussion](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N034OGrEu5o4a_TlhHL:-N034OGrEu5o4a_TlhHM:b7csuvk) (no related file):*
review required .

---


<!-- Sent from Reviewable.io -->

nbali committed:
Merge branch 'master' into execution_config_factory_test

nbali reviewed (commented):
> Reviewed 3 of 3 files at r22, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 1 unresolved discussion (waiting on @csabakassai, @gaborvolczer, and @kollerdaniel)



<!-- Sent from Reviewable.io -->

nbali commented:
@gaborvolczer will you be able to review it?

nbali committed:
Merge branch 'master' into execution_config_factory_test

nbali reviewed (commented):
> Reviewed 1 of 1 files at r23, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 1 unresolved discussion (waiting on @csabakassai, @gaborvolczer, and @kollerdaniel)



<!-- Sent from Reviewable.io -->

nbali commented:
@gaborvolczer ping

nbali reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: all files reviewed, 1 unresolved discussion (waiting on @csabakassai, @gaborvolczer, and @kollerdaniel)

*[a discussion](https://reviewable.io/reviews/aliz-ai/jarvis/41#-N034OGrEu5o4a_TlhHL:-N1iV-2o8hENbslLhdp7:b-s1khs7) (no related file):*
<details><summary><i>Previously, nbali (Balázs Németh) wrote…</i></summary><blockquote>

review required .
</blockquote></details>

@gaborvolczer 

___


<!-- Sent from Reviewable.io -->

nbali reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/aliz-ai/jarvis/41)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @csabakassai and @gaborvolczer)



<!-- Sent from Reviewable.io -->
